subprojects/common/report-viewer/src/main/kotlin/com/avito/report/internal/JsonRpcRequestProvider.kt

```diff
@@ -20,21 +20,44 @@ internal class JsonRpcRequestProvider(
 
     inline fun <reified T : Any> batchRequest(request: List<RfcRpcRequest>): T = internalRequest(request)
 
-    private inline fun <reified Response : Any> internalRequest(request: Any): Response {
-        val response = httpClient.newCall(
-            Request.Builder()
-                .url(host.removeSuffix("/"))
-                .header("Accept", jsonMime)
-                .header("Content-Type", jsonMime)
-                .post(gson.toJson(request).toRequestBody(jsonMime.toMediaType()))
-                .build()
-        ).execute()
+    private inline fun <reified Response : Any> internalRequest(jsonRpcRequest: Any): Response {
```

Comment:
Propose to replace `Any` with the concrete type

Reply:
`Any` was chosen, because `Any` is a parameter of `gson.toJson()`, and it was only consumer

Reply:
I don't understand. Could you explain this decision wider?
What I see, we have a concrete type for jsonRpcRequest but decide to cast it to `Any` because of JSON
I don't understand what problems JSON has with concrete type